code/modules/clothing/under/marine_uniform.dm

```diff
@@ -527,23 +527,56 @@
 	min_cold_protection_temperature = ICE_PLANET_MIN_COLD_PROT
 	has_sensor = UNIFORM_HAS_SENSORS
 	sensor_faction = FACTION_UPP
-	suit_restricted = list(/obj/item/clothing/suit/storage/marine/faction/UPP, /obj/item/clothing/suit/gimmick/jason, /obj/item/clothing/suit/storage/snow_suit/soviet, /obj/item/clothing/suit/storage/snow_suit/survivor)
+	suit_restricted = list(/obj/item/clothing/suit/storage/marine/faction/UPP, /obj/item/clothing/suit/gimmick/jason, /obj/item/clothing/suit/storage/snow_suit/soviet, /obj/item/clothing/suit/storage/snow_suit/survivor, /obj/item/clothing/suit/storage/webbing)
 	flags_jumpsuit = UNIFORM_SLEEVE_ROLLABLE
 
+/obj/item/clothing/under/marine/veteran/UPP/survivor
+
+/obj/item/clothing/under/marine/veteran/UPP/survivor/Initialize(mapload)
+	. = ..()
+	var/R = rand(1,2)
+	switch(R)
+		if(1)
+			roll_suit_sleeves(TRUE)
+		if(2)
+			roll_suit_jacket(TRUE)
+
```

Comment:
Creating a subtype just to randomise if the suit is rolled in places or not is a little excessive/bloat.

Reply:
Is there an alternative way for me to get the uniforms to start with the sleeves or jacket rolled?

Reply:
In load_gear() of the gear set create the object, do your random check, change it on the uniform, then equip it.

Reply:
Morrow, you might one day make a coder out of me.